.github/workflows/nightly.yml

```diff
@@ -15,10 +15,7 @@ jobs:
       fail-fast: false
       matrix:
         os: ["ubuntu-latest", "macos-14", "windows-latest"]
-        python-version: ["3.9", "3.11"]
-        exclude:
-        - os: "macos-14"
-          python-version: "3.9"  # not available for macos-14
+        python-version: ["3.10", "3.11", "3.12"]
```

Comment:
let's not test 3.11 here? If things work for 3.10 and 3.12 they should also work for 3.11. And having 3.11 here will cost a bunch of GHA capacity (note that macos hosts are more expensive)